src/debugpy/common/sockets.py

```diff
@@ -9,18 +9,60 @@
 from debugpy.common import log
 from debugpy.common.util import hide_thread_from_debugger
 
+def can_bind_ipv4_localhost():
+    """Check if we can bind to IPv4 localhost."""
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        # Try to bind to IPv4 localhost on port 0 (any available port)
+        sock.bind(('127.0.0.1', 0))
+        sock.close()
+        return True
+    except (socket.error, OSError):
```

Comment:
Any other errors that should be captured? Otherwise, this function may return `None`.

Reply:
LGTM